js/sign/src/utils/utils.ts

```diff
@@ -31,15 +36,63 @@ export function parsePemKey(
   });
 }
 
-export function getRawPublicKey(publicKey: crypto.KeyObject) {
-  // Currently this is the only way for us to get the raw 32 bytes of the public key.
-  return new Uint8Array(
-    publicKey.export({ type: 'spki', format: 'der' }).slice(-32)
+export function isAsymmetricKeyTypeSupported(key: crypto.KeyObject) {
+  return (
+    key.asymmetricKeyType === 'ed25519' ||
+    (key.asymmetricKeyType === 'ec' &&
+      key.asymmetricKeyDetails!.namedCurve === 'secp256k1')
```

Comment:
nit (same behavior, but without the risk to ever attempt to access `namedCurve` on `undefined`): 
```suggestion
      key.asymmetricKeyDetails?.namedCurve === 'secp256k1')
```

Reply:
Done!